### PR TITLE
Don't emit stderr on successful bootstrap.

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

**Review by ras0219-msft — APPROVED** [review]

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -119,7 +119,7 @@ vcpkgDownloadFile()
     url=$1; downloadPath=$2 sha512=$3
     vcpkgCheckRepoTool "curl"
     rm -rf "$downloadPath.part"
-    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" || exit 1
+    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" --silent --show-error || exit 1
```
Comment (ras0219-msft) [inline scripts/bootstrap.sh:122]:
```suggestion
    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" --silent --show-error --fail || exit 1
```
Is `--fail` appropriate here?

Comment (brycelelbach) [reply scripts/bootstrap.sh:122]: Hrm good question - `--silent --show-error` was my suggestion, let me research what `--fail` does.

**Comment by brycelelbach** [comment]
Context: it can be annoying to have tools output to stderr in non-error cases, because some consoles/UIs highlight errors in a special way. You don't want attention called too non-error cases. The unfortunate alternative is to redirect stderr to stdout, but then real errors don't get highlighted.

This came up when I was adding vcpkg into the Docker images we use for github.com/NVIDIA/thrust and github.com/NVIDIA/cub

bootstrap_vcpkg_sh_curl_stderr

File: scripts/bootstrap.sh (hunk 1 above)
Comment (brycelelbach) [reply scripts/bootstrap.sh]: I'm hesitant to suggest the use of `--fail`. The `curl` docs say:

> `-f, --fail          Fail silently (no output at all) on HTTP errors`

The reason for `--silent --show-error` is to get no output except in the case of errors. `--silent` means no output, and `--show-error` means show errors even in silent mode.

Comment (brycelelbach) [reply scripts/bootstrap.sh]: @BillyONeal says it changes "http 404 + here's a document explaining that it was a 404" into "no file and return an error code", so it seems good.

**Pull request merged**